runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -126,8 +132,11 @@ public CheckpointWriter(CorfuRuntime rt, UUID streamId, String author, T map) {
      */
     public Token appendCheckpoint() {
         long start = System.currentTimeMillis();
+        Token markerToken = rt.getSequencerView().next().getToken();
```

Comment:
Just use the tail directly, this is not needed.

`Token markerToken = rt.getSequencerView().query(streamId)`

Reply:
Cool. And will read on the stream tail so as to persist it by hole-filling if needed.